### ISSUE TITLE
Some questions to understand the model better

### ISSUE THREAD

**Issue by smoosavi** [KaiyangZhou/deep-person-reid#158]
I would appreciate it if you could clarify them for me as I am very new in your model:
1. In your model what is the input and what is the output?
2. How are these images or tackles being labeled? 
3. How does the model work? What goes in as input and how it selects the outputs in case of a ResNet50 backbone? How does the process of selection and detection happens?
4. What about the loss functions?
5. How is the label smoothing involve here?
Thank you so much.

**Comment by KaiyangZhou** [comment]
> In your model what is the input and what is the output?

The input to a model is image data with the batch form `(batch_size, num_channel, height, width)`; the output is image descriptors with shape `(batch_size, feature_dimension)`.

> How are these images or tackles being labeled?

Each image/tracklet is associated with a person ID label. A better example is to go through a dataset description like [Market1501](https://jingdongwang2017.github.io/Projects/ReID/Datasets/Market-1501.html).

> How does the model work? What goes in as input and how it selects the outputs in case of a ResNet50 backbone? How does the process of selection and detection happens?

Please refer to answer1. Current reid datasets typically contain cropped person images so it's essentially a retrieval task. The detection is used as a preprocessing step to obtain the person images.

> What about the loss functions?

Most methods use either margin-based loss (e.g. [triplet loss](https://arxiv.org/abs/1703.07737)), classification loss (cross entropy), or a combination of them.

Margin-based loss compares a pair of images and aims to push the images away from each other in the feature space if they contain different persons or pull them together if they contain the same person. You could also find some papers where they compare more than two images (e.g. [quadruplet loss](https://arxiv.org/abs/1704.01719))

Classification loss formulates the learning process as a classification task where the goal is to classify each training image into one of the known identities. This is similar to the classic image classification pipeline.

> How is the label smoothing involve here?

Just like the motivation when label smoothing is proposed, it's used here to regularize the person ID classification by introducing soft labels, e.g. `probability=0.9` for true class and `probability=0.1` for false class. It's particularly useful in the context of person reid because person images can be similar in appearance even if they contain different persons.

**Comment by smoosavi** [comment]
Thank you for the clarifications. I have one last question, when the system is for example, up to Rank-5, one image query and 5 gallery images will go to the model for classifications??

**Comment by KaiyangZhou** [comment]
I guess you are confused about the classification in person reid. The pipeline is exactly the same as the normal classification, i.e. you give the network an image, the network makes its prediction on which person class (identity) it belongs to, and then the cross entropy loss is computed.

**Comment by smoosavi** [comment]
Thank you. Yes, I might have, I guess my confusion comes from what happens at each rank. For example in rank 5 what is differing than rank 10 in the process of classification and generating the final ranking list?

**Comment by KaiyangZhou** [comment]
The classification pipeline is for training purpose only. It might be a bit confusing at the first time as person reid is a retrieval task where retrieval loss is usually adopted. But it turned out that classification is a very strong learning pipeline, by which the learned feature embedding is discriminative for person identities.

The final ranking list is obtained by comparing the feature distance between query and gallery images.

**Comment by smoosavi** [comment]
Thank you so much :)
